app/models/proxy_config.rb

```diff
@@ -37,21 +37,20 @@ class InvalidEnvironmentError < StandardError; end
   scope :newest_first,   -> { order(version: :desc) }
   scope :by_environment, ->(env) { where(environment: VALID_ENVIRONMENTS[env]) }
   scope :by_host,        ->(host) { where.has { hosts =~ "%|#{host}|%" } if host }
-
-  scope :current_versions, -> do
-    table = BabySqueel[:proxy_configs].alias(:versions)
-    scope = joining { table.on((table.proxy_id == proxy_id) & (table.environment == environment)) }
-      .when_having { max(table.version) == version }
-      .group(:id)
-
-    System::Database.mysql? ? scope : where(id: scope.group(:version).select(:id))
-  end
-
-  scope :by_version, ->(version) do
-    next unless version
-    next current_versions if version == 'latest'
-
-    where(version: version)
+  scope :by_version,     ->(version) { where(version: version) if version }
+  scope :by_proxy_ids,   ->(proxy_ids) { where.has { proxy_id.in(proxy_ids) } if proxy_ids }
+
+  def self.latest_versions(environment:, proxy_ids: nil)
+    where.has do
+      id.in(
+        ProxyConfig
+          .selecting { MAX(id).as('id') } # There is only 1 proxy_config with the maximum version for the same proxy_id and environment
```

Comment:
What if `proxy_ids` is `nil`? We get the max version across different services?

Reply:
That's not possible.
https://github.com/3scale/porta/blob/2d3d1f33ffe144b8e4f0006fdbf22e988341fae3/db/postgres_schema.rb#L1071
https://github.com/3scale/porta/blob/2d3d1f33ffe144b8e4f0006fdbf22e988341fae3/db/schema.rb#L1071
https://github.com/3scale/porta/blob/2d3d1f33ffe144b8e4f0006fdbf22e988341fae3/db/oracle_schema.rb#L1122